src/main.rs

```diff
@@ -208,6 +209,47 @@ fn build_nmap_arguments<'a>(
     arguments
 }
 
+/// Goes through all possible IP inputs (files or via argparsing)
+/// Parses the string(s) into IPs
+fn parse_addresses(opts: &Opts) -> Vec<IpAddr> {
+    // Construct a new Resolver with default configuration options, DNS.
+    let mut ips: Vec<IpAddr> = Vec::new();
+
+    for ip_or_host in &opts.addresses {
+        match read_ips_from_file(ip_or_host.to_owned()) {
+            Ok(x) => ips.extend(x),
+            _ => match parse_single_ip_or_host(ip_or_host.to_owned()) {
+                Ok(x) => ips.extend(x),
+                _ => {
+                    warning!(
+                        format!("Host {:?} could not be resolved.", ip_or_host),
+                        opts.greppable,
+                        opts.accessible
+                    );
+                }
+            },
+        }
+    }
+    println!("Ips is {:?}", ips);
+    ips
+}
+
+/// Uses DNS to get the IPS assiocated with host
+fn get_ips_from_host(ip_or_host: &String) -> Result<Vec<IpAddr>, std::io::Error> {
+    let mut ips: Vec<std::net::IpAddr> = Vec::new();
+    let resolver = Resolver::new(ResolverConfig::default(), ResolverOpts::default()).unwrap();
```

Comment:
I dislike how every time this function is called, I create a new resolver. I couldn't get it to work via arguments (it doesn't implement a lot of traits and `cargo check` hated it.

The alternative is a global variable, unless someone can make a further PR fixing this :)

Reply:
Never mind! This might have been a different bug and I confused the two. Seems to be working just fine now :)